### PR TITLE
Rename "group" to "circle"

### PR DESCRIPTION
**Pull request by MrPetovan**
This is part one of addressing #13114, first replacing "group" in the "contact group" sense by "circle". The only remaining occurrences are regarding the database tables that I wouldn't dare renaming without involving @annando first.

This branch is currently running on my dev node without a hitch, I'll let it run some more just to be sure.

### CODE REVIEW

**Comment by annando** [comment]
I just returned from my weekend trip. It can take several days until I catch up with all the changes. We shouldn't - in no way - alter table or field names. (I haven't looked at the PR)

File: doc/Quick-Start-andfinally.md
Comment (Quix0r) [inline doc/Quick-Start-andfinally.md:5]: Is here a word missing?

File: doc/de/Account-Basics.md
Comment (Quix0r) [inline doc/de/Account-Basics.md:112]: You can translate these all to "Kreise". Nextcloud does this, too.

File: doc/de/Circles-and-Privacy.md
Comment (Quix0r) [inline doc/de/Circles-and-Privacy.md:1]: Same here, "Kreise" is okay.

File: doc/de/Home.md
Comment (Quix0r) [inline doc/de/Home.md:17]: Same here, "Kreise" is okay.

File: src/Module/Admin/Site.php
Comment (Quix0r) [inline src/Module/Admin/Site.php]: Please fix indenting.

File: src/Module/Api/Friendica/Circle/Create.php
Comment (Quix0r) [inline src/Module/Api/Friendica/Circle/Create.php:81]: Logic and renaming in one go. Is this now the correct way?

File: src/Module/Contact.php
Comment (Quix0r) [inline src/Module/Contact.php]: Please fix indenting.

File: src/Module/Contact.php
Comment (Quix0r) [inline src/Module/Contact.php]: Can we indent this, too? So that the code is aligned after the `=` sign?

File: src/Module/Search/Acl.php
Comment (Quix0r) [inline src/Module/Search/Acl.php:280]: Good find!

File: src/Module/Settings/Account.php
Comment (Quix0r) [inline src/Module/Settings/Account.php:595]: Here a change from `displayGroupSelection()` to `getSelectorHTML()` has happened. I lost a bit the oversight during review here. Is this the correct new name?

File: src/Module/User/Import.php
Comment (Quix0r) [inline src/Module/User/Import.php:234]: Maybe change this comment by adding a `@todo` to it as one day this all can be removed?

File: src/Object/Api/Friendica/Circle.php
Comment (Quix0r) [inline src/Object/Api/Friendica/Circle.php:30]: Class documentation has been removed here.

File: src/Protocol/ActivityPub/Receiver.php
Comment (Quix0r) [inline src/Protocol/ActivityPub/Receiver.php:1101]: Please double-check this change from `$actor` to `$profile`.

File: src/Security/Security.php
Comment (Quix0r) [inline src/Security/Security.php]: `$g` comes from group, if we follow naming-convention, then this should be renamed at least to `$c` or even `$circle`.

File: static/defaults.config.php
Comment (Quix0r) [inline static/defaults.config.php:316]: Is there a backward-compatible option (deprecated of course) still possible? This might break user's setup.

File: static/routes.config.php
Comment (Quix0r) [inline static/routes.config.php:99]: Also these comments may be marked with `@todo` or `@deprecated`?

File: static/settings.config.php
Comment (Quix0r) [inline static/settings.config.php:119]: Same here: Is a backward-compatible option possible?

**Review by Quix0r — CHANGES_REQUESTED** [review]
Maybe some parts need to be changed.

File: doc/Quick-Start-andfinally.md
Comment (MrPetovan) [reply doc/Quick-Start-andfinally.md:5]: In this file, groups means forums.

File: src/Module/Settings/Account.php
Comment (MrPetovan) [reply src/Module/Settings/Account.php:595]: Yes, the previous name was redundant and not very informative, the new name is closer to the naming convention I like to go for.

File: src/Object/Api/Friendica/Circle.php
Comment (MrPetovan) [reply src/Object/Api/Friendica/Circle.php:30]: Yeah, it wasn't adding any meaningful details.

File: static/settings.config.php
Comment (MrPetovan) [reply static/settings.config.php:119]: Yes, we will first check the value of `compute_group_counts` and then `compute_circle_counts`.

File: static/defaults.config.php
Comment (MrPetovan) [reply static/defaults.config.php:316]: Yes, we will check the value of `groupedit_image_limit` first, then `circle_edit_image_limit`.

File: src/Security/Security.php
Comment (MrPetovan) [reply src/Security/Security.php]: Good catch!

File: src/Protocol/ActivityPub/Receiver.php
Comment (MrPetovan) [reply src/Protocol/ActivityPub/Receiver.php:1101]: `$actor` is a URL string coming from the function parameter, it can't be used to access a `type` key. On the other hand, this key is present in the return value of `APContact::getByURL`.

File: src/Module/Api/Friendica/Circle/Create.php
Comment (MrPetovan) [reply src/Module/Api/Friendica/Circle/Create.php:81]: Sure, the change is minor.

File: src/Module/User/Import.php
Comment (MrPetovan) [reply src/Module/User/Import.php:234]: This is a hard one, user archives will stay with the wrong keys long after the node that generated it has upgraded, so we can't really remove this code ever.

File: doc/de/Account-Basics.md
Comment (MrPetovan) [reply doc/de/Account-Basics.md:112]: I'm not comfortable making this change on my own, a few German people said "Circle" was fine.

Comment (nupplaphil) [reply doc/de/Account-Basics.md:112]: I would prefer Circles too, Kreise seems too much direct translated into German.

**Comment by MrPetovan** [comment]
Can we look at this PR so that I can move on to replacing "forum" by "group"?

**Review by Quix0r — APPROVED** [review]

File: doc/de/Account-Basics.md
Comment (AndyHee) [reply doc/de/Account-Basics.md:112]: I'm surprised that "Circles" is the preferred choice. Although I have near-native speaking ability, I wouldn't exactly know what German people say.

Either way, I'm puzzled about the plural of "Circles" in German. Does it not follow the logic of "Zirkel" (singular) -  "Zirkel" (plural)? Is that how people really talk?

Considering the fact that Nextcloud uses the German term (Kreis), I'd say there should be a discussion  (with a consensus or vote based outcome) in the German-speaking community on the exact translation.

Comment (Quix0r) [reply doc/de/Account-Basics.md:112]: Don't we use the word Freundeskreis anymore?

Comment (MrPetovan) [reply doc/de/Account-Basics.md:112]: @AndyHee There was a discussion, and you contributed to it: https://github.com/friendica/friendica/issues/13114#issuecomment-1546939430

Comment (AndyHee) [reply doc/de/Account-Basics.md:112]: This might not really be the place here to have that discussion. Possibly better to talk about it in the [Support forum](https://forum.friendi.ca/profile/helpers) in German.

But on a sidenote, I once heard an older German gentleman saying: "_In welchen Kreisen verkehren Sie eigentich?_" Which means something like 'In what circles are you socialising?'. Apparently it's a polite way of expressing disapproval. 😲

Comment (AndyHee) [reply doc/de/Account-Basics.md:112]: @MrPetovan Sorry I mean a discussion about the correct German translation. 

Are you worried that it will be hijacked and turned into a discussion of whether to change the actual terminology or not? 

(I'm only seeing your reply now. Why?)

Comment (AndyHee) [reply doc/de/Account-Basics.md:112]: Maybe for practical purposes, let's just push this through and revisit the translation issue later?

As said, I'm surprised about the choice of translation. But I'm not on that translation team and my knowledge is limited.

How to proceed? @MrPetovan @nupplaphil @Quix0r?

Comment (MrPetovan) [reply doc/de/Account-Basics.md:112]: I'm disappointed you are surprised because your participated in the issue discussion where we talked about it and I received two separate confirmations by German people in the same conversation that "Circle" was fine for the German documentation.

The worst that could happen right now is if this PR is blocked by yet another language debate after I sought and received an answer and put work to it.

Comment (AndyHee) [reply doc/de/Account-Basics.md:112]: Yes, I noticed that some German people wanted to settle on "Circle". Today I saw what this really looks like. Hence, I added my comment.

I think you're right. It's not wise to mix these discussions. I understand that you moved the German documentation into this PR for swiftly moving this forward. But it might have contributed to the little difficulty. 

So, let push this through, and we'll leave it to the German translation team to review it.

I'm sorry that you felt disappointed.

Comment (MrPetovan) [reply doc/de/Account-Basics.md:112]: The issue here is that the German documentation pages aren't part of the regular translation workflow, meaning that developers have to make these decisions instead of the translators over at Transifex. I'm not sure if there's a way to include the documentation pages in Transifex, nor if it's desirable.

Comment (AndyHee) [reply doc/de/Account-Basics.md:112]: Sure, I probably would have done exactly the same. Let's not worry too much about this.

**Comment by annando** [comment]
Is this whole "how to translate it into German" solved? I hadn't had a deep look at the discussion.

**Comment by AndyHee** [comment]
**Yes**, the issue is solved. We agreed that discussions about translations should not interfere with completing this important PR.

The proposed changes to the German documentation are _sufficient_ for merging with the development branch now.

If deemed necessary, German-speaking users can discuss the changes to the documentation once the PR has merged. There will be sufficient time that allows for community participation, prior to the 09.2023 stable release.

File: README.md
Comment (annando) [inline README.md:22]: What are the spaces here for? Is this intentional?

Comment (MrPetovan) [reply README.md:22]: phpStorm has its own code formatting rules when it comes to Markdown tables.

It caused me problems with the database documentation 😅

However, these changes don’t affect the HTML display.

Comment (annando) [reply README.md:22]: Have you changed the database documentation by hand? You know that calling `bin/console dbstructure dumpsql > database.sql` does not only create the `database.sql` file but does also update the database documentation?

**Comment by annando** [comment]
Only two things here:
1. Had you performed a test run (1 day or so) on your productive system and hat a look at the logs?
2. Please fix the conflict.

File: README.md
Comment (MrPetovan) [reply README.md:22]: Yes but the system where I edit the code (local laptop) isn’t the same than the system where I have the database (remote VPS) so I thought I could get away with string replacement. In the end I had to use dumpsql but it isn’t convenient.

**Comment by MrPetovan** [comment]
I have run this code on my dev node, clicking around the circles (formerly groups) pages successfully.

**Comment by MrPetovan** [comment]
This branch is now running on my production node.

File: static/routes.config.php
Comment (Quix0r) [inline static/routes.config.php]: Indenting is missing here.

**Review by Quix0r — CHANGES_REQUESTED** [review]

**Comment by annando** [comment]
Okay, then it is now duck, cover and having a look for some unexpected behaviour :grimacing:

**Comment by MrPetovan** [comment]
No, the next PR is coming which restores "group" but under a different meaning now!

**Pull request merged**
